### PR TITLE
Stp 833 default govuk notify email

### PR DESCRIPTION
**Pull request by bertrandmc**
# Description

STP-833-default-govuk-notify-email.

- Make Output by email the default option. 
- Refactors and TSify outputs related code and moves files to `/outputs` folder.
- Adjust model workplace types built to emit alongside code instead of separated types folder (TS wasn't recognizing types in Designer). 

## Aceptance criteria. 

As a user
Given I am editing a form
When I click `Edit Outputs`
And click `Add output`
Then I can see that `Output Type` is `Email` by default

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [X] I have added tests that prove my changes are effective 
- [X] New and existing unit tests pass locally with my changes

# Checklist:

- [X] My changes do not introduce any new linting errors
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation and versioning
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have rebased onto master and there are no code conflicts

### CODE REVIEW

File: designer/client/outputs/output-edit.tsx [hunk 1]
```diff
@@ -1,49 +1,75 @@
...
       outputIndex = data.outputs.indexOf(output);
     }
 
-    let outputConfiguration = output.outputConfiguration || {};
+    let outputConfiguration: OutputConfiguration =
+      output.outputConfiguration || {};
+
     switch (outputType) {
-      case "notify":
+      case OutPutType.Email:
```
Comment (jenbutongit) [inline designer/client/outputs/output-edit.tsx:55]: being pedantic, sorry! output is one word so `outputType`

File: designer/client/outputs/output-edit.tsx [hunk 2]
```diff
@@ -1,49 +1,75 @@
...
         };
         break;
-      case "email":
+      case OutPutType.Notify:
         outputConfiguration = {
-          emailAddress: formData.get("email-address"),
+          personalisation: formData
+            .getAll("personalisation")
+            .map((t) => (t as string).trim()),
+          templateId: formData.get("template-id") as string,
```
Comment (jenbutongit) [inline designer/client/outputs/output-edit.tsx:65]: I assume we can get rid of type casts when we type the data object?

File: designer/nodemon.json [hunk 3]
```diff
@@ -1 +1,4 @@
-{}
+{
+  "delay": "2500",
+  "watch": ["dist"]
+}
```
Comment (jenbutongit) [inline designer/nodemon.json:4]: 🙏

File: model/tsconfig.json [hunk 4]
```diff
@@ -3,9 +3,10 @@
   "baseUrl": "./src",
   "compilerOptions": {
     "outDir": "dist",
-    "rootDirs": ["src"],
-    "declarationDir": "dist/types",
-    "composite": true
+    "rootDir": "src",
+    // "declarationDir": "dist/types",
```
Comment (jenbutongit) [inline model/tsconfig.json:7]: is this meant to be commented out?

File: model/tsconfig.tsbuildinfo [hunk 5]
```diff
@@ -0,0 +1,2498 @@
+{
```
Comment (jenbutongit) [inline model/tsconfig.tsbuildinfo:1]: does this need to be gitignored?

**Review by jenbutongit — APPROVED** [review]

**Review by jenbutongit — CHANGES_REQUESTED** [review]
oops sorry, I hit approve by accident.. please see (probably dismissed comments)

File: designer/client/outputs/notify-edit-items.tsx [hunk 6]
```diff
@@ -1,28 +1,39 @@
...
     this.state = {
       items: props.items ? clone(props.items) : [],
     };
   }
 
-  onClickAddItem = (e) => {
-    this.setState({
-      items: this.state.items.concat(""),
-    });
+  onClickAddItem = (_event: MouseEvent) => {
```
[the remark this answers is not in the document]
Comment (bertrandmc) [reply designer/client/outputs/notify-edit-items.tsx:23]: I don't know, I am not refactoring the code at that lower level, the purpose of this refactor is to TSify, I might find minor things and fix them, mostly when TS complains about something.

File: designer/client/outputs/output-edit.tsx (hunk 2 above)
Comment (bertrandmc) [reply designer/client/outputs/output-edit.tsx:65]: Yep!

File: model/tsconfig.json (hunk 4 above)
Comment (bertrandmc) [reply model/tsconfig.json]: Removed. Thanks

File: model/tsconfig.tsbuildinfo (hunk 5 above)
Comment (bertrandmc) [reply model/tsconfig.tsbuildinfo]: Yes, ignored. Thanks

**Review by jenbutongit — APPROVED** [review]

File: designer/client/outputs/output-edit.tsx (hunk 1 above)
Comment (bertrandmc) [reply designer/client/outputs/output-edit.tsx]: Not pedantic, wrong is wrong. Thanks

**Pull request merged**